### ISSUE TITLE
Add description for each target

### ISSUE THREAD

**Issue by russcam** [adamralph/bullseye#525]
## Use case(s)

Ability to describe each target when specifying `--list-targets`.

## Description

When listing the targets, it would be useful to see a short description that describes what each target does. This would allow a user to get a better understanding without needing to look through the source of each target. The target names should hopefully give some indication, and a description would be able to add further detail.

For example, in one current project, listing the targets with `.\build.bat --list-targets` writes

```
agent-zip
build
clean
default
diff
netcore-sln
pack
publish
release-notes
restore
version
```

With descriptions, this list would be more useful to a user

```
agent-zip:		builds a zip file of all assemblies needed to instrument with `DOTNET_STARTUP_HOOKS`
build:			builds all projects
clean:			cleans all projects
default:		runs the build target
diff:			performs a diff on each assembly to understand what has changed
netcore-sln:		creates a solution file containing only .NET Core projects
pack:			creates nuget packages for all publishable projects
publish:		publishes all publishable projects
release-notes:		generates github release notes based on issue and PR labels
restore:		restores all nuget package dependencies
version:		configures the version to run with.
```

## Alternatives

Descriptions and further details could be provided through a chosen command line integration e.g. `System.CommandLine`, and additional details provided through this, for example, defining sub commands. There might be some difficulty in this approach however in terms of the applicability of arguments in certain contexts - As an example, if a sub command `diff` is configured with `System.CommandLine` and the command `build.bat diff --list-dependencies` is run, `System.CommandLine` expects `--list-dependencies` to be a valid argument in the context of the sub command.

## Possible implementation(s)

- `Target` has a `string Description {get;}` property, which can be set with the ctor
- Methods for being able to specify the properties of a target such as `Targets.Target(...)` and overloads, accept an additional parameter for the description.

**Comment by adamralph** [comment]
Hi @russcam, thanks for raising this. I was waiting for someone to request this. 😉 

I'm not opposed to it, but one problem it does raise is adding yet another optional parameter to the `Targets.Target()` method group, which means yet another set of overloads. As you know, Bullseye doesn't have a fluent API, nor does it use optional parameters. However, both of those design choices can be revisited. In order to implement this, we have to carefully consider the shape of that method group and how best to add this new parameter.

**Comment by russcam** [comment]
I had a little look through the issues and was surprised to see no one had raised already 😄 

Agree that additions need to be done judiciously. A fluent API might be reasonable, and offers some flexibility particularly if there are other optional properties to a Target that may be introduced later e.g. condition(s). I personally like the relative simplicity of Bullseye, which I _think_ is intentional?

**Comment by adamralph** [comment]
I'm glad to hear you like the relatively simplicity and yes, it's very much intentional. 🙂 I'd like to avoid introducing a fluent API if possible.

**Comment by russcam** [comment]
Thanks for adding in #556, @adamralph 👍

**Comment by adamralph** [comment]
@russcam you're welcome. I've just released this in [3.7.0-alpha.1](https://www.nuget.org/packages/Bullseye/3.7.0-alpha.1). Would you like to give it a try?

**Comment by russcam** [comment]
@adamralph thanks for the shoutout! ❤️ I'll give it a try now
